### ISSUE TITLE
Gpu Memory Leak in the Nodes forward Method 

### ISSUE THREAD

**Issue by ArefAz** [BindsNET/bindsnet#551]
Hi, 

I'm using Bindsnet to build a brain-inspired image classifier. When I trained the model on my data for lots of iterations, I noticed that my GPU memory was getting fully utilized until the code crashed. I pinpoint where the problem is coming from and that was 
`self.v += (self.refrac_count <= 0).float() * x` part of the code in the forward pass of `DiehlAndCookNodes` class. I also tested the code with the `BoostedLIFNodes` class, and the same problem showed up in the line `self.v += x`. 

After a bit of research, I found that the problem is due to the fact that PyTorch is normally storing the gradients for this type of operation, which I believe is not required in Bindsnet for unsupervised learning (Please correct me if I'm wrong). What fixed the issue for me was just adding the `with torch.no_grad():` before the actual operation.

I suggest adding it to the code base so this problem won't happen in the future for others.

Thank you so much!

**Comment by Hananel-Hazan** [comment]
Thanks @ArefAz for lets us know. Certainly interesting findings and for this simple fix.  

@sidharthanup FYI.

**Comment by sidharthanup** [comment]
@ArefAz Can you let me know the pytorch version you are using?

**Comment by ArefAz** [comment]
@sidharthanup Sure. It is v1.10.0.
